cpp/CONTRIBUTING.md

```diff
@@ -0,0 +1,170 @@
+# Bulding the C++ Cookbook
+
+The C++ cookbook combines output from a set of C++ test programs with
+an RST document tree rendered with sphinx.
+
+Running `make py` from the cookbook root directory (the one where
+the `README.rst` exists) will install all necessary dependencies,
+run the tests to generate the output, and will compile the cookbook
+to HTML.
+
+You will see the compiled result inside the `build/cpp` directory.
+
+The above process requires conda to be installed and is primarily
+intended for build systems. See below for more information on setting
+up a development environment for developing recipes.
+
+# Developing C++ Recipes
+
+Every recipe is a combination of prose written in **reStructuredText**
+format using the `Sphinx <https://www.sphinx-doc.org/>`\_ documentation
+system and a snippet of a googletest test.
```

Comment:
nits: if we spell out reStructuredText here, maybe we should do so above?

Should we link to reST and Googletest since we're also linking to Sphinx?